client/src/components/employer-dashboard-component/employer-dashboard-filter/index.jsx

```diff
@@ -9,8 +9,19 @@ import {
 import SearchIcon from "@mui/icons-material/Search";
 import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
 import { LocalizationProvider, DatePicker } from "@mui/x-date-pickers";
+import PropTypes from "prop-types";
+import CardActions from "@mui/material/CardActions";
+import Button from "@mui/material/Button";
+import * as React from "react";
+import JobLeadType from "../../../prop-types/JobLeadType";
 
-function EmployerDashboardFilter() {
+function EmployerDashboardFilter({ paginationModel, handleApplyFilter }) {
+  // setting and persisting initial state
+  const [noFilterMode, setNoFilterMode] = React.useState(true);
+  const [ignorePaginationChange, setIgnorePaginationChange] =
```

Comment:
Could you give a bit of an explanation how and why these two states are being used?

Reply:
Yeah so before i explain these two here is how i general filter and pagination work: I have a useEffect within the filter component that triggers whenever pagination changes.

The reason being is the case where we set a filter (say filter June 1st) and the user might want to change the page to say page 2. From this, I have a useEffect that triggers whenever the paginationModel (this case change to page 2) within the filter component so that it invokes a fetch with the state of the filters that we had already (filter June 1st) AND the updated pagination model so that we reinvoke an api to get query for the same filter but this time different page.

So thats the main functionality.

Now onto the where the 2 useStates that maybe unclear is used. 

1. ignorePaginationChange: there is also a case where the user wants to apply a new filter WHILE on another page (say page 2). We want it so that whenever a user applies a new filter then we reset to page back to page 1. That means we need to invoke a change to the pagination AND THEN a fetch with our new filter queries. But remember how I explained before we have a useEffect that triggers whenever pagination changes, this will cause an infinite loop. So how can I reset the pagination back to page 1 and then invoke a fetch with the new filter query without triggering another fetch because of my useEffect I used before? I have a binary toggle, ignorePaginationChange that lets the useEffect know when to run and when to ignore.

2. noFilterMode is solely on initial render to make sure that the initial fetch will not have any queries so that means we know we are fetching all of the employers.

Reply:
Let me know if it may still be unclear, would be happy to hop in a call.